http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -173,8 +156,63 @@ void sendMessage(final String message) {
       if (!connectionIsOpen) {
         queuedMessages.add(message);
       } else {
-        client.send(message);
+        client
+            .sendText(message, true)
+            .exceptionally(logWebSocketError(Level.SEVERE, "Failed to send text."));
       }
     }
   }
+
+  private <T> Function<Throwable, T> logWebSocketError(
+      final Level level, final String errorMessage) {
+    return throwable -> {
+      log.log(level, errorMessage, throwable);
+      return null;
+    };
+  }
+
+  @VisibleForTesting
+  class WebSocketListener implements Listener {
+    private final StringBuilder textAccumulator = new StringBuilder();
+
+    @Override
+    public void onOpen(final WebSocket webSocket) {
+      synchronized (queuedMessages) {
+        client = webSocket;
+        connectionIsOpen = true;
+        queuedMessages.forEach(
+            message ->
+                client
+                    .sendText(message, true)
+                    .exceptionally(logWebSocketError(Level.SEVERE, "Failed to send queued text.")));
+        queuedMessages.clear();
+      }
+      webSocket.request(1);
+    }
+
+    @Override
+    public CompletionStage<?> onText(
+        final WebSocket webSocket, final CharSequence data, final boolean last) {
+      textAccumulator.append(data);
```

Comment:
It takes a lot of convincing that this is threadsafe. Messages coming in will be concurrent, ie: multiple chatters to server and those being handled on different threads then sending back messages concurrently to this listener. 

Perhaps a documentation reference and some commentary on why we are guaranteed to be threadsafe here would be useful for the future. A naive maintainer might look at this and decide we *must* have a sync lock or start to think the implementation is broken. It seems like an obvious problem if two 'onText' events come in at the same time and they are both not `last`, then we'll mix the messages together.

Reply:
1. If that helps the code is almost 1:1 taken from an example implementation from the documentation: https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/WebSocket.html#request(long) so this is the way it's meant to be used.
2. I haven't actually thought of that and I see how it bothers you. I'll add some documentation which should hopefully clarify that.